src/scenes/AcceleratorRouter/Participants/ParticipantForm.tsx

```diff
@@ -0,0 +1,204 @@
+
+import AddLink from 'src/components/common/AddLink';
+import Card from 'src/components/common/Card';
+import PageForm from 'src/components/common/PageForm';
+import { useAcceleratorOrgs } from 'src/hooks/useAcceleratorOrgs';
+import { useCohorts } from 'src/hooks/useCohorts';
+import strings from 'src/strings';
+import { AcceleratorOrg } from 'src/types/Accelerator';
+import { Cohort } from 'src/types/Cohort';
+import { ParticipantCreateRequest, ParticipantUpdateRequest } from 'src/types/Participant';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+
+import OrgProjectsSectionEdit, { OrgProjectsSection } from './OrgProjectsSectionEdit';
+
+type ParticipantFormProps<T extends ParticipantCreateRequest | ParticipantUpdateRequest> = {
+  busy?: boolean;
+  participant: T;
+  onCancel: () => void;
+  onSave: (participant: T) => void;
+};
+
+export default function ParticipantForm<T extends ParticipantCreateRequest | ParticipantUpdateRequest>(
+  props: ParticipantFormProps<T>
+): JSX.Element {
+  const { busy, participant, onCancel, onSave } = props;
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+  const { availableCohorts } = useCohorts();
+  const { acceleratorOrgs: allAcceleratorOrgs } = useAcceleratorOrgs();
+
+  const [localRecord, setLocalRecord] = useState<T>(participant);
+  const [validateFields, setValidateFields] = useState<boolean>(false);
+  // initialize with one org selection enabled
+  const [orgProjectsSections, setOrgProjectsSections] = useState<OrgProjectsSection[]>([
+    { id: 1, selectedProjectIds: [] },
+  ]);
+  // orgs available for selection in the dropdowns, does not included already selected orgs
+  const [availableOrgs, setAvailableOrgs] = useState<AcceleratorOrg[]>([]);
+
+  const acceleratorOrgs = useMemo<AcceleratorOrg[]>(
+    () => (allAcceleratorOrgs || []).filter((org) => org.availableProjects.length > 0),
+    [allAcceleratorOrgs]
+  );
+
+  const cohortOptions = useMemo(
+    () => (availableCohorts || []).map((cohort: Cohort) => ({ label: cohort.name, value: cohort.id })),
+    [availableCohorts]
+  );
+
+  const updateField = useCallback((field: keyof T, value: any) => {
+    setLocalRecord((prev) => ({
+      ...prev,
+      [field]: value,
+    }));
+  }, []);
+
+  const validateForm = () => {
+    if (!localRecord.name) {
+      return false;
+    }
+
+    if (!localRecord.cohort_id) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const onSaveHandler = () => {
+    if (!validateForm()) {
+      setValidateFields(true);
+      return;
+    }
+
+    onSave({
+      ...localRecord,
+    });
+  };
+
+  const onOrgSelect = useCallback(
+    (sectionId: number, orgId: number) => {
+      const org = acceleratorOrgs?.find((o) => o.id === orgId);
+      if (!org) {
+        return;
+      }
+      setOrgProjectsSections((prev) => {
+        const updated = [...prev];
+        updated[sectionId - 1].org = org;
```

Comment:
I get why this works, but is a bit tricky. I guess the alternative is either passing the index through (not great) or `[...prev].map(_return original if ID is different, otherwise update_)`. Anyways, not a big deal, but perhaps a comment as to why the id is being used as an index might be helpful.

Reply:
I can use the index approach and remove 'id' semantics since it might be easier to read.
I didn't get the `[...prev].map(...` suggestion, can you clarify? If that's better I'd be up for using it.

Reply:
```
const updated = [...prev].map((section) => section.id !== sectionId 
  ? section 
  : {
    ...section,
    org,
    selectedProjectIds: []
  }
```

Reply:
Ah ok, instead of an index access, thanks. I will use this and add a comment on the 'id'. Realized using an index will remove the ability to use the 'useCallback' efficiently.

Reply:
Thanks for the suggestion, definitely cleaner and hides the index into array semantics.